Source/Core/Core/Boot/Boot.cpp

```diff
@@ -272,7 +272,8 @@ bool CBoot::BootUp()
 
     const DiscIO::IVolume& pVolume = DVDInterface::GetVolume();
 
-    if ((pVolume.GetVolumeType() == DiscIO::Platform::WII_DISC) != _StartupPara.bWii)
+    if (pVolume.GetVolumeType() == DiscIO::Platform::WII_DISC &&
+        _StartupPara.m_system != SConfig::System::Wii)
```

Comment:
Yes, that's what the check seems to be for. I just found it a bit confusing to compare two bools like that directly (`is_wii_disc != is_wii_console`), so I changed that to `is_wii_disc && !is_wii_console`